File: Programmers/Lv2/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    # 각 귤의 갯수를 센다
    gull = {}

    for g in tangerine:
        if g in gull:
            gull[g] += 1
        else:
            gull[g] = 1

    # 제일 많은 종류를 넣는다

    tmp = sorted(gull.items(), key=lambda x: x[1], reverse=True)
    cnt = 0
    i = 0
    while k > 0:
        cnt += 1
        k -= tmp[i][1]
        i += 1
    answer = cnt
    return answer

File: Programmers/Lv2/귤 고르기.js
function solution(k, tangerine) {
    var answer = new Array();
    var dic = new Map();
    for(num of tangerine){
        dic.set(num, (dic.get(num) || 0) + 1)
    }
    let tmp = [...dic].sort((a, b) => b[1] - a[1]);
    
    while(k > 0){
        let now = tmp[0][1];
        answer.push(tmp[0][0]);
        k -= now;
        tmp.shift()
    }
    return answer.length;
}